go/types/value_encoder.go

```diff
@@ -26,8 +27,10 @@ func (w *valueEncoder) writeKind(kind NomsKind) {
 
 func (w *valueEncoder) writeRef(r Ref) {
 	w.writeHash(r.TargetHash())
-	w.writeType(r.TargetType(), map[string]*Type{})
-	w.writeUint64(r.Height())
+	if !w.forRollingHash {
```

Comment:
The rolling has encodes values and passes them to buzhash. The only sense in which this affects the hash is where the chunk boundary in drawn.